### PR TITLE
Reorganizes wideq.py into different files.

### PR DESCRIPTION
**Pull request by boralyl**
The idea here is to be able to break up the big wideq.py into different files to make it easier to manage. This is also preparation for adding more device types like washer/dryer.  This will also make testing a bit easier.

This doesn't change any code (except for the one comment below), just moves classes into different files.

### CODE REVIEW

File: wideq/device/__init__.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            self.device.id,
+            key,
+            'Control',
+        )
+
+        # The response comes in a funky key/value format: "(key:value)".
+        _, value = data[1:-1].split(':')
+        return value
+
+    def monitor_start(self):
```
Comment (boralyl) [inline wideq/device/__init__.py]: This is the only code change, moved `monitor_start` and `monitor_end` into the base Device class.  Looking at your `ACDevice` and wkd8176's fork with washer/dryer/etc. devices, these 2 methods are always identical, so it makes more sense to put this in the base class.

File: wideq/__init__.py [hunk 2]
```diff
@@ -4,12 +4,15 @@
...
 import uuid
 import base64
-import json
 import hashlib
 import hmac
 import datetime
-from collections import namedtuple
-import enum
+
+# These imports are for backwards compatiability with previous versions.
```
Comment (boralyl) [inline wideq/__init__.py]: Importing everything here so moving these classes does't cause any compatibility issues with previous versions.

**Comment by sampsyo** [comment]
Awesome; thanks for continuing this! It looks, however, like the testing stuff snuck back into this PR somehow?

As for the reorganization, I actually had a slightly different logical split in mind, if you're amenable to it:

- `__init__` or `core` or `base`: Everything in the current source file up to `Session`. This is a low-level, procedural interface that encodes the basic protocol that the server uses to communicate. This would probably include the exceptions, unless they're more appropriately located elsewhere.
- `client`: At least the `Client` and `Monitor` classes, and probably also `Device` and its associated code. This is a higher-level abstraction that code should use to ignore the low-level details and provides convenient utilities like serialization. I sort of see `Client` and `Device` as a single piece that works together.
- `ac`: All the current AC-specific stuff. I don't think we need a subdirectory for different device types; this can just be `wideq.ac` at the top level.

Re-exporting everything at the top level makes a lot of sense to me!

**Comment by boralyl** [comment]
I couldn't choose a branch other than master.  This is a branch off of the prior 2 pr's, so it includes all changes

**Comment by boralyl** [comment]
> Awesome; thanks for continuing this! It looks, however, like the testing stuff snuck back into this PR somehow?
> 
> As for the reorganization, I actually had a slightly different logical split in mind, if you're amenable to it:
> 
>     * `__init__` or `core` or `base`: Everything in the current source file up to `Session`. This is a low-level, procedural interface that encodes the basic protocol that the server uses to communicate. This would probably include the exceptions, unless they're more appropriately located elsewhere.
> 
>     * `client`: At least the `Client` and `Monitor` classes, and probably also `Device` and its associated code. This is a higher-level abstraction that code should use to ignore the low-level details and provides convenient utilities like serialization. I sort of see `Client` and `Device` as a single piece that works together.
> 
>     * `ac`: All the current AC-specific stuff. I don't think we need a subdirectory for different device types; this can just be `wideq.ac` at the top level.
> 
> 
> Re-exporting everything at the top level makes a lot of sense to me!

This sounds reasonable to me

**Comment by sampsyo** [comment]
Oops; looks like we hit a race condition! I implemented the above proposal. Feel free to take a look at how I did it.

**Comment by boralyl** [comment]
Heh, I wasn't sure so I went ahead and made the changes, but ultimately they look nearly identical to your commit.  Thanks!

**Pull request closed without merging**